### ISSUE TITLE
[$2000] IOU - Pay via PayPal appears only after refreshing the page after adding PayPal in the sender

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#14935]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login in 2 accounts at the same time (A and B)
2. As account A sent Request money to account B
3. As account B opens the chat and clicks on the pay selector dropdown (the PayPal option is absent)
4. As account A go to Settings -> Payments -> Add PayPal method
5. Verify that Pay via PayPal appears

## Expected Result:
Pay via PayPal is present

## Actual Result:
Pay via PayPal appears only after refreshing the page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.66.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/217406200-66de4f6c-867d-4565-a58a-3a04b7387dbf.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012081ec54cd4071b8</li>
        <li>Upwork Job ID: 1625071482929115136</li>
        <li>Last Price Increase: 2023-02-20</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @zanyrenney (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Current assignee @zanyrenney is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by zanyrenney** [comment]
Reviewed and applied labels.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @flodnv (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by abdulrahuman5196** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We are not seeing Pay vai paypal option without manually refreshing the page

### What is the root cause of that problem?
We use submitterPayPalMeAddress value from the IOUReport date to decide on whether to show the pay via paypal option or not. 
https://github.com/Expensify/App/blob/main/src/pages/iou/IOUDetailsModal.js#L188
When the requestor updates the paypal address we don't get any pusher event or any update so the UI doesn't update. We re-fetch the data when we reload the page which results in latest paypal data being propagated.

### What changes do you think we should make in order to solve the problem?
This is mostly likely a backend change. Backend needs to send us pusher event incase of the paypal removal/addition updates to have the data updated properly in the client UI.


### What alternative solutions did you explore? (Optional)
None

**Comment by PankajAS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Payment options are not updating when money request sender remove payment option

### What is the root cause of that problem?
This is problem with same time update payment option while User B try to send money and User A remove select payment method.

### What changes do you think we should make in order to solve the problem?
We have to check selected payment method is still valid or not if its not there anymore then it should show error and update options because as you can see if i close and open IOUDetailsModal then its showing me updated options.

### What alternative solutions did you explore? (Optional)
As a alternative option we have to update options real time IOUDetailsModal which will be not right at the moment.

**Comment by zanyrenney** [comment]
Hey @abdulrahuman5196 @PankajAS - thanks so much for your proposals. 

@flodnv what are your thoughts on these proposals?

**Comment by flodnv** [comment]
I agree @aimane-chnaif. I think we close this as won't fix, it's incredibly corner case and a simple refresh fixes it.

**Comment by flodnv** [comment]
Probably an API error, which hopefully gets shown in a Red Brick Road fashion...

**Comment by zanyrenney** [comment]
Cool, thanks @flodnv, deleted the upwork posting and  - closing!
